types/src/validator_set.rs

```diff
@@ -36,6 +37,21 @@ pub(crate) fn validator_set_path() -> Vec<u8> {
 #[cfg_attr(any(test, feature = "testing"), derive(Arbitrary))]
 pub struct ValidatorSet(Vec<ValidatorPublicKeys>);
 
+impl fmt::Display for ValidatorSet {
+    fn fmt(&self, f: &mut fmt::Formatter) -> std::fmt::Result {
+        // I still don't understand why Rust doesn't want me to be able to display vectors,
+        // but whatever.
```

Comment:
this is probably a reasonable answer - https://internals.rust-lang.org/t/impl-t-display-display-for-option-t/6051/2